src/main/kotlin/org/rust/ide/annotator/RsMutableSelfParameterAnnotator.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator
+
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.psi.PsiElement
+import org.rust.ide.colors.RsColor
+import org.rust.lang.core.psi.RsPath
+import org.rust.lang.core.psi.RsSelfParameter
+import org.rust.lang.core.psi.ext.mutability
+import org.rust.lang.core.types.ty.Mutability
+
+class RsMutableSelfParameterAnnotator : Annotator {
+    private val LOW_PRIORITY_HIGHLIGHTING = HighlightSeverity(
```

Comment:
I think in our case annotation priority of this annotator should be greater than `RsHighlightingAnnotator`, i.e. greater than `HighlightSeverity.INFORMATION` but less than `HighlightSeverity.WEAK_WARNING`